### PR TITLE
Set default volume on audiostream creation

### PR DESCRIPTION
**Pull request by the-maddin**
Addressing my issue #3225, I'm setting sane values found in other
parts of the sound system at stream creation, to be used or overwritten later.

Also fixed a potential leak of stream slots and deduplicated some code.

### CODE REVIEW

**Comment by the-maddin** [comment]
Also updated scripting documentation, because that's what `audiostream_play()` _actually_ does.

**Comment by Goober5000** [comment]
Hmm.  Who is familiar with the audio code?  @ln-zookeeper or @z64555 or @notimaginative perhaps?

File: code/sound/audiostr.cpp [hunk 1]
```diff
@@ -922,6 +923,47 @@ void audiostream_close()
...
+            break;
+        case ASF_VOICE: // As in: sound.cpp:590
+            Audio_streams[i].Set_Default_Volume(Master_voice_volume * aav_voice_volume);
+            break;
+        default:
+            Audio_streams[i].status = ASF_FREE;
+            return -1;
+    }
+
+    Audio_streams[i].Set_Default_Volume(type == ASF_VOICE ? Master_voice_volume * aav_voice_volume : Master_sound_volume);
```
Comment (asarium) [inline code/sound/audiostr.cpp:962]: This will overwrite the value set previously. Is that intended?

**Review by asarium — CHANGES_REQUESTED** [review]
Please also make sure to use the proper whitespace configuration (tabs over spaces). We have an editorconfig file which is supported by most editors (either natively or via a plugin).

File: code/sound/audiostr.cpp (hunk 1 above)
Comment (the-maddin) [reply code/sound/audiostr.cpp:962]: That was an oversight, left in from before I realized there are more than the two defaults exposed via LUA. Correcting.

**Comment by notimaginative** [comment]
LGTM, other than the whitespace issue that asarium pointed out.

**Review by asarium — APPROVED** [review]

**Pull request merged**